diff -rupN old_player-3.0.2/server/drivers/shell/readlog.cc player-3.0.2/server/drivers/shell/readlog.cc
--- old_player-3.0.2/server/drivers/shell/readlog.cc	2009-12-15 22:51:26.000000000 -0200
+++ player-3.0.2/server/drivers/shell/readlog.cc	2015-07-18 22:54:53.200586631 -0300
@@ -665,7 +665,7 @@ void ReadLog::Main()
       // back up to the beginning of the file
 #if HAVE_Z
       if (this->gzfile)
-        ret = gzseek(this->file,0,SEEK_SET);
+        ret = gzseek(this->gzfile,0,SEEK_SET);
       else
         ret = fseek(this->file,0,SEEK_SET);
 #else
@@ -711,7 +711,7 @@ void ReadLog::Main()
       // compared to fgets (on uncompressed files), so use the latter.
 #if HAVE_Z
       if (this->gzfile)
-        ret = (gzgets(this->file, this->line, this->line_size) == NULL);
+        ret = (gzgets(this->gzfile, this->line, this->line_size) == NULL);
       else
         ret = (fgets(this->line, this->line_size, (FILE*) this->file) == NULL);
 #else
